### PR TITLE
fix: ArgumentOutOfRangeException when adding PrimaryCommand to NavBar

### PR DESCRIPTION
**Pull request by Kunal22shah**
GitHub Issue (If applicable): #1150 

## PR Type

What kind of change does this PR introduce?

- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by kazo0** [comment]
@Kunal22shah make sure to use the keyword `closes` or `fixes` before the link to the affected issue number:

image

If you change it to 

> closes #1150

Then the PR will be linked to the issue and it will auto-close the issue when this PR is merged

File: src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarPresenter.cs
Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarPresenter.cs:1]: @Xiaoy312 Just double checking that this looks good? I feel like I stole the old code from somewhere so there must be somewhere else that has this code wrong. Maybe here? https://github.com/unoplatform/uno/blob/c7485a6415ffb5c783950a6f562ea9ec1f0798f4/src/Uno.UI/UI/Xaml/Controls/CommandBar/CommandBar.Partial.cs#L1410-L1411

File: src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarPresenter.cs [hunk 1]
```diff
@@ -227,32 +227,41 @@ private void OnNavBarSecondaryCommandsChanged(IObservableVector<ICommandBarEleme
 
 		private void OnCommandsChanged(IObservableVector<ICommandBarElement> sender, IVectorChangedEventArgs args, DependencyProperty prop)
 		{
-			if (_commandBar != null)
+			if (_commandBar == null) return;
+			var change = args.CollectionChange;
+			var changeIndex = args.Index;
+			var commands = _commandBar.GetValue(prop) as IObservableVector<ICommandBarElement>;
+			if (commands == null) return;
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarPresenter.cs:234]:
```suggestion
			if (_commandBar.GetValue(prop) is not IObservableVector<ICommandBarElement> commands) return;
```

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarPresenter.cs
Comment (Xiaoy312) [reply src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarPresenter.cs:1]: looks about right~

**Pull request merged**